ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift

```diff
@@ -26,17 +25,23 @@ import UIKit
 
     @objc
     public func createPaywallView() -> PaywallViewController {
-        let controller = PaywallViewController()
+        let controller = PaywallViewController(dismissalRequest: { controller in
+            self.delegate?.paywallViewControllerRequestedDismissal?(controller)
+        })
         controller.delegate = self
-        
         return controller
     }
 
     @objc
-    public func createPaywallView(offeringIdentifier: String) -> PaywallViewController {
-        let controller = PaywallViewController(offeringIdentifier: offeringIdentifier)
+    public func createPaywallView(
+        offeringIdentifier: String,
+        dismissHandler: (() -> Void)? = nil
+    ) -> PaywallViewController {
+        let controller = PaywallViewController(offeringIdentifier: offeringIdentifier,
+                                               dismissalRequest: { controller in
```

Comment:
We need to update the name of this parameter and add it to the footer methods... Maybe we can even extract to a function to avoid repeating the code?

Reply:
Though thinking about it, we might need to make `self` weak here I think?